src/main/groovy/wooga/gradle/unity/tasks/ProjectManifestTask.groovy

```diff
@@ -0,0 +1,49 @@
+package wooga.gradle.unity.tasks
+
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurper
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import org.gradle.process.ExecResult
+import wooga.gradle.unity.UnityTask
+import wooga.gradle.unity.models.ResolutionStrategy
+import wooga.gradle.unity.traits.UnityPackageSpec
+
+abstract class ProjectManifestTask extends DefaultTask implements
+    UnityPackageSpec {
+
+    final static String manifestFileName = "manifest.json"
+    final static String lockFileName = "packages-lock.json"
+
+    abstract void onExecute(Object manifest)
+
+    @TaskAction
+    void execute() {
+        modifyProjectManifest(new Action<Object>() {
+            @Override
+            void execute(Object o) {
+                onExecute(o)
+            }
+        })
+    }
+
+    private Boolean modifyProjectManifest(Action<Object> onParse) {
```

Comment:
just call it action and do not use the `on...` naming scheme. It is not a handler of event system. We generally don't use these naming patterns.

Reply:
Hmm, sure, though I do not think the `onX` implies an event system, but that of a callback.